src/components/Attachments/AttachmentCarousel/index.tsx

```diff
@@ -88,10 +89,16 @@ function AttachmentCarousel({report, reportActions, parentReportActions, source,
             return;
         }
 
-        const initialPage = targetAttachments.findIndex(compareImage);
+        let initialPage = targetAttachments.findIndex(compareImage);
+        const prevInitialPage = attachments.findIndex(compareImage);
+
+        // If no matching attachment is found in targetAttachments but found in attachments, update initialPage
+        if (initialPage === -1 && prevInitialPage !== -1 && targetAttachments[prevInitialPage]) {
```

Comment:
Can we avoid comparing with -1 and instead just access the attachment array directly

```js
if (!targetAttachments[attachmentIndex] && targetAttachments[prevAttachmentIndex])
```

Reply:
done